kettle/make_json.py

```diff
@@ -157,13 +160,17 @@ def row_for_build(path, started, finished, results):
         if 'version' in finished:
             build['version'] = finished['version']
 
-    def get_metadata():
+    def get_metadata(build):
         metadata = None
+        metapairs = None
         if finished and 'metadata' in finished:
             metadata = finished['metadata']
         elif started:
             metadata = started.get('metadata')
+
         if metadata:
+            if not build.get('repos'):
+                build['repos'] = metadata.get('repos')
```

Comment:
updating `build` in a function named `get_metadata` feels surprising, maybe we could do this after the `get_metadata` call below?

Reply:
Yeah ^ above comment kinda covers it. Just feels a little ugly to iterate over the list of pairs. I could return both the list AND the repos from get_metadata and assign that second value to `build['repos']`. Also in your doc you mention that Repos needs to be a nullable string. Do I need to join these repos with commas?

Reply:
I majorly refactored. I decided to stay on the safe side and just remove all keys for `None` attrs so that there isn't anything crazy that might happen with nullable fields. I think this cleans things up and creates a structure for the build object so it is easier to map to the TestGrid Schema